haystack/nodes/preprocessor/base.py

```diff
@@ -10,7 +11,7 @@ class BasePreProcessor(BaseComponent):
     @abstractmethod
     def process(
         self,
-        documents: Union[dict, List[dict]],
+        documents: Union[dict, Document, List[Union[dict, Document]]],
```

Comment:
Maybe we can in future only support List instead of single items here? Seems unintuitive to me that `documents` can be a single `dict`.

Reply:
I added a DeprecationWarning in the `PreProcessor`'s `process` method that is triggered if the user does not supply a list.